src/terms/literal.rs

```diff
@@ -0,0 +1,203 @@
+        assert_eq!(expected, l1);
+    }
+
+    #[test]
+    fn display_for_english_slice() {
+        let expected = Literal {
+            value: "foo",
+            language: None,
+            data_type: NamedNode {
+                value: "http://www.w3.org/2001/XMLSchema#string".to_string(),
+            },
+        };
+
+        let l1 = Literal::new("foo").unwrap();
+        assert_eq!(expected, l1);
+    }
+
+    #[test]
+    fn display_for_spanish_string() {
+        let expected = Literal {
+            value: "Hola Mundo",
+            language: Some("es".to_string()),
+            data_type: NamedNode {
+                value: "http://www.w3.org/2001/XMLSchema#string".to_string(),
+            },
+        };
+
+        // let l1 = Literal::for_language("Hola Mundo", "es");
+        let l2 = Literal::new("Hola Mundo")
+            .unwrap()
+            .with_language("es")
+            .unwrap();
+
+        assert_eq!(expected, l2);
+        // todo!();
+    }
+
+    #[test]
+    fn display_for_integer() {
+        let expected = Literal {
+            value: 42,
+            language: None,
+            data_type: NamedNode {
+                value: "http://www.w3.org/2001/XMLSchema#integer".to_string(),
+            },
+        };
+
+        let l1 = Literal::new(42).unwrap();
+        assert_eq!(expected, l1);
+    }
+
+    #[test]
+    fn display_for_decimal() {
+        let expected = Literal {
+            value: 42.42,
+            language: None,
+            data_type: NamedNode {
+                value: "http://www.w3.org/2001/XMLSchema#double".to_string(),
+            },
+        };
+
+        let l1 = Literal::new(42.42).unwrap();
+        assert_eq!(expected, l1);
+    }
+
+    #[test]
+    fn display_for_boolean() {
+        let expected = Literal {
+            value: true,
+            language: None,
+            data_type: NamedNode {
+                value: "http://www.w3.org/2001/XMLSchema#boolean".to_string(),
+            },
+        };
+
+        let l1 = Literal::new(true).unwrap();
+        assert_eq!(expected, l1);
+    }
+
+    #[test]
+    fn display_for_datetime() {
+        let expected = Literal {
+            value: Utc::today().and_hms(0, 0, 0),
+            language: None,
+            data_type: NamedNode {
+                value: "http://www.w3.org/2001/XMLSchema#dateTime".to_string(),
+            },
+        };
+
+        let l1 = Literal::new(Utc::today().and_hms(0, 0, 0)).unwrap();
+        assert_eq!(expected, l1);
+    }
+
+    #[test]
+    fn display_for_date() {
+        let expected = Literal {
+            value: Utc::today(),
+            language: None,
+            data_type: NamedNode {
+                value: "http://www.w3.org/2001/XMLSchema#date".to_string(),
+            },
+        };
+
+        let l1 = Literal::new(Utc::today()).unwrap();
+        assert_eq!(expected, l1);
+    }
+
+    #[test]
+    fn display_for_time() {
```

Comment:
As far as I can tell, all of these `display...` tests only test that the constructor (and the `with_language` method) are working as expected, as you never use `format!("{literal}")` or `literal.to_string()` (which is automatically implemented when you implement `Display`). Maybe I'm misunderstanding what you mean by `display_for_x`?

Reply:
good catch - renamed for next commit and that's a good reminder that I still need to implement the display trait, since there is some real formatting consideration